internal/flags/helpers.go

```diff
@@ -213,3 +229,44 @@ func wordWrap(s string, width int) string {
 
 	return output.String()
 }
+
+// AutoEnvVars extens all the specific CLI flags with automatically generated
+// env vars by capitalizing the flag, replacing . with _ and prefixing it with
+// the specified string.
+//
+// Note, the prefix should *not* contain the separator underscore, that will be
+// added automatically.
+func AutoEnvVars(flags []cli.Flag, prefix string) {
+	for _, flag := range flags {
+		envvar := strings.ToUpper(prefix + "_" + strings.ReplaceAll(strings.ReplaceAll(flag.Names()[0], ".", "_"), "-", "_"))
+
+		switch flag := flag.(type) {
+		case *cli.StringFlag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+
+		case *cli.BoolFlag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+
+		case *cli.IntFlag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+
+		case *cli.Uint64Flag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+
+		case *cli.DurationFlag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+
+		case *cli.PathFlag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+
+		case *BigFlag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+
+		case *TextMarshalerFlag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+
+		case *DirectoryFlag:
+			flag.EnvVars = append(flag.EnvVars, envvar)
+		}
```

Comment:
This is super ugly, good luck making it nicer :D

Reply:
This will become much easier with cli.v3 I think, because it uses generics to define the flag types.

Reply:
Could always use generics here.

Reply:
Meh :) Not worth the complexity for fixing 20 lines of code.